src/model/consent/index.ts

```diff
@@ -0,0 +1,61 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+/*
+ * Class to encapsulate Consent functions for DB dependencies
+ */
+import Knex from 'knex'
+import {
+  Consent,
+  registerConsent,
+  updateCredentialsByConsentId,
+  getConsentById,
+  deleteConsentById
+} from './consent'
+
+export default class ConsentModel {
```

Comment:
Let migrate `ConsentModel` class from src/model/consent/index to external file, move all db logic from functions in src/model/consent/consent to `ConsentModel` methods - there is no need to keep Object approach and Functional approach in the same time...

Simplify the names of methods to keep the code compact. There is no need to repeat `Consent` section in methods names. If you return Promise value the method should have `async` prefix

```
const consent = {...}
const consentModel = new ConsentModel(db)

// instead of 
consentModel.registerConsent(consent)

// use
consentModel.register(consent)
```

Reply:
Where do you suggest I move the ConsentModel class along with the logic? Do you have a new directory structure in mind?

Reply:
Addressed in a recent commit.